### PR TITLE
adds the ability to list scheduled print jobs

### PR DESCRIPTION
**Pull request by cameroncros**

### CODE REVIEW

File: octoprint_discordremote/command_plugins/printscheduler.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        builder.set_author(name=self.plugin.get_printer_name())
+
+        for job in data:
+            title = ("Name: %s" % job['name'])
+            description = job['start_at']
+
+            builder.add_field(title=title, text=description)
+
+        # Use data in status to build an embed
+        return None, builder.get_embeds()
```
Comment (cameroncros) [inline octoprint_discordremote/command_plugins/printscheduler.py:37]: I notice this is incorrect for a few other plugins, but it should be:
```suggestion
        return builder.get_embeds()
```

Comment (cameroncros) [reply octoprint_discordremote/command_plugins/printscheduler.py:37]: image

Comment (cameroncros) [reply octoprint_discordremote/command_plugins/printscheduler.py:37]: @jneilliii

**Comment by jneilliii** [comment]
Thanks, I was basing the structure on enclosure. I also noticed you're using global api key. That will be going away in favor of application keys. One of the reasons I thought it would be better to not use api calls.

**Comment by cameroncros** [comment]
Yeah, the entire code base is bitrotting away because I don't have time to
maintain it :( I'll have to fix the broken code eventually when it fail.

**Comment by cameroncros** [comment]
Still has issues, can't remove job after adding it:
image

Also, the removejob function tends to silently swallow a bunch of errors when the arguments don't match correctly, Ideally, they should be handled and return an error message to discord:
image

**Comment by jneilliii** [comment]
Yeah, I wasn't fully finished with the changes. I did have one question [here](https://github.com/cameroncros/OctoPrint-DiscordRemote/issues/208#issuecomment-973381755) in regard to entering parameters with spaces. The timestamp format is very specific though `2021-11-18 22:00`

**Comment by cameroncros** [comment]
If I recall correctly, if the timestamp is quoted, it should split the
parameters correctly.

**Comment by jneilliii** [comment]
ok I'll do some additional debugging then because maybe I'm not defining them properly. Couldn't find an example with multiple parameters, only single ones in the existing codebase and SgtKiLLx said it wasn't working to add jobs either. Will need to build a discord server to figure it out though I suppose.

**Comment by cameroncros** [comment]
I have a discord server you can use for development, and I can give you the
details if you want them.

https://discord.gg/Ws8B7VHc <- If you click that link, you can join the
test server, and then I will give you the bot token and channel id so that
you can test.

**Comment by jneilliii** [comment]
no worries, I didn't realize it was that easy to setup. I was able to get it to work but had to rearrange the parameters basically because nothing I tried would allow spaces. Not that big of a deal though.

image

**Pull request merged**